### PR TITLE
[SUBMISSION] MODULE 1

### PR DESCRIPTION
**Pull request by Knight7561**
# December 2024 Student Submission

## Module Completed
- [x] Module 1: Instruction Tuning
- [ ] Module 2: Preference Alignment
- [ ] Module 3: Parameter-efficient Fine-tuning
- [ ] Module 4: Evaluation
- [ ] Module 5: Vision-language Models
- [ ] Module 6: Synthetic Datasets
- [ ] Module 7: Inference
- [ ] Module 8: Deployment

## Changes Made
Describe what you've done in this PR:
1. What concepts did you learn?
- Tokenisation, Reading datasets from HF, fine tuning smol models and how tokinzer module is contained
2. What changes or additions did you make?
- Added comments for better understanding, will add additional examples in the future work
3. Any challenges you faced?
- Nothing for this module

## Notebooks Added/Modified
List any notebooks you've added or modified:
- [x] Added new example in `module_name/student_examples/my_example.ipynb`
- [ ] Modified existing notebook with additional examples
- [ ] Added documentation or comments

## Checklist

- [x] I have read the module materials
- [x] My code runs without errors
- [x] I have pushed models and datasets to the huggingface hub
- [x] My PR is based on the `december_2024` branch

## Questions or Discussion Points
Add any questions you have or points you'd like to discuss:
1. 
2. 

## Additional Notes
Any other information that might be helpful for reviewers:

### CODE REVIEW

**Comment by burtenshaw** [comment]
Structurally this looks great. 

@ShankarChavan Could you peer-review the notebooks please?

**Comment by Knight7561** [comment]
@burtenshaw @ShankarChavan Is my PR good ?

**Review by ShankarChavan — CHANGES_REQUESTED** [review]
In your **SFT_finetuning_example** the model has not shown any improvement on loss function. Also, the response from model for the given test-questions were giving repeated response as model has not learned yet. Please try to fix these issues from your end.

**Review by burtenshaw — APPROVED** [review]
Nice work @Knight7561
